psi4/driver/procrouting/scf_proc/scf_iterator.py

```diff
@@ -270,25 +312,43 @@ def scf_iterate(self, e_conv=None, d_conv=None):
     efp_enabled = hasattr(self.molecule(), 'EFP')
     cosx_enabled = "COSX" in core.get_option('SCF', 'SCF_TYPE')
 
-    # does the JK algorithm use severe screening approximations for early SCF iterations?
-    early_screening = False
+    # does the SCF process require two "stages" of SCF iterations?
+    # if so, are we on the first round of SCF iterations?
+    # formalism currently applies to the following:
+    #    - COSX: initial grid -> final grid 
+    #    - COSX_SCF_GUESS: COSX -> DirectJK/DF-DirJ+LinK
+    #    NOTE: DF_SCF_GUESS fits the criteria here, but does not explicitly utilize this formalism 
+    on_first_iter_round = False
     if cosx_enabled:
-        early_screening = True
+        on_first_iter_round = True
         self.jk().set_COSX_grid("Initial")
 
-    # maximum number of scf iterations to run after early screening is disabled
-    scf_maxiter_post_screening = core.get_option('SCF', 'COSX_MAXITER_FINAL')
+    # are we doing are final round of SCF iterations?
+    on_final_iter_round = False
+    # maximum number of scf iterations to run during the final round of SCF iterations, for applicable methods 
+    scf_maxiter_final_round = core.get_option('SCF', 'COSX_MAXITER_FINAL')
+
+    # account for DirectJK/DF-DirJ+LinK iterations post-COSX-guess if need be
+    if core.get_option('SCF', 'SCF_COSX_GUESS') and any([ core.get_global_option('SCF_TYPE') == x for x in [ 'DIRECT', 'DFDIRJ+LINK' ] ]):
+        on_first_iter_round = False 
+        on_final_iter_round = True
 
-    if scf_maxiter_post_screening < -1:
-        raise ValidationError('COSX_MAXITER_FINAL ({}) must be -1 or above. If you wish to attempt full SCF converge on the final COSX grid, set COSX_MAXITER_FINAL to -1.'.format(scf_maxiter_post_screening))
+        # two DirectJK/DF-DirJ+LinK iterations unless specified
+        if not core.has_option_changed("SCF", "COSX_MAXITER_FINAL"):
+          scf_maxiter_final_round = 2 # run two iterations with COSX guess
+        else:
+          scf_maxiter_final_round = core.get_option("SCF", "COSX_MAXITER_FINAL") # run COSX_MAXITER_FINAL # of DirectJK iterations with COSX guess
```

Comment:
Most simply, the settings I selected were tested to give (~0.1 kcal/mol) accuracy for a variety of relative energies on systems I tested.

I came at this PR from the angle of doing a COSX calculation, where the final COSX grid iteration(s) is replaced by DirectJK iteration(s). Since COSX in Psi4 uses a single SCF iteration on the final grid by default, that would carry over here too, given how I am approaching this PR. I just had to bump the DirectJK iteration count up from 1 to 2 to make the accuracy acceptable, as per above.

You could also look at it as a `DF_SCF_GUESS`-type calculation, where you use a small COSX grid instead of MemDF for the initial part of the calculation, which is seemingly (and understandbaly) how you are approaching this PR. 

In my experience with `COSX_MAXITER_FINAL=-1`, it can take a noticeable amount of SCF iterations to converge the second part of the calculation, especially with such a small initial grid as we use here, so setting `COSX_MAXITER_FINAL` to 2 here avoids that.

Reply:
Thanks for the additional information. Is "noticeable" more like 5 or 10 (essentially a full set) iterations?

What do others think? I'm so used to SCF as prelude to post-SCF that I probably have a skewed opinion on SCF as-target.

Reply:
Yeah, "noticeable" is in the 5-10 range.